app/services/payout_report_publisher_includer.rb

```diff
@@ -0,0 +1,57 @@
+class PayoutReportPublisherIncluder < BaseService
+  def initialize(payout_report:, publisher:, should_send_notifications:)
+    @publisher = publisher
+    @payout_report = payout_report
+    @should_send_notifications = should_send_notifications
+  end
+
+  def perform
+    return if @publisher.suspended? || !@publisher.has_verified_channel?
+    publisher_has_unsettled_balance = false
+
+    wallet = @publisher.wallet
+    return if wallet.nil?
+
+    probi = wallet.channel_balances[@publisher.owner_identifier].probi_before_fees # probi = balance
+    if probi > 0
+      publisher_has_unsettled_balance = true
+
+      if @publisher.uphold_verified? && wallet.address.present?
+        PotentialPayment.create(payout_report_id: @payout_report.id,
+                                name: @publisher.name,
+                                amount: "#{probi}",
+                                fees: "0",
+                                publisher_id: @publisher.id,
+                                kind: PotentialPayment::REFERRAL,
+                                address: "#{wallet.address}")
+      end
+    end
+
+    @publisher.channels.verified.each do |channel|
+      probi = wallet.channel_balances[channel.details.channel_identifier].probi # probi = balance - fee
+      next if probi <= 0 || !@publisher.uphold_verified? || wallet.address.blank?
+
+      publisher_has_unsettled_balance = true
+      fee_probi = wallet.channel_balances[channel.details.channel_identifier].fee # fee = balance - probi
+
+      PotentialPayment.create(payout_report_id: @payout_report.id,
```

Comment:
I'm trying to understand the logic a little bit more, why do we create two potential payouts, one for just the publisher and then one for all the publishers channels?

I'm curious, if the publisher doesn't have upload verified should we be preventing these payouts from being created?

Reply:
This is due to eyeshade accounting logic - referrals accrue to to the publisher whereas contributions accrue to the channel. This way if a channel is moved any unsettled balance moves with it.